wordpress-mobile/WordPress-Aztec-iOS
# Propagating visual-newline deletion to the DOM: part #1

<h3>Description:</h3>

Takes care of properly propagating the deletion of certain visual-newlines to the DOM structure.  This does NOT cover all cases of visual newlines, but should provide a baseline for dealing with it.

Solves some of #289 

<h3>Testing:</h3>

**Test 1:**

Review and run the added unit tests.

**Test 2:**

1. Open the empty demo.
2. Switch to HTML mode.
3. Introduce `<p>hello</p>there` (or any variation of two block-level elements where one or both is a paragraph).
4. Switch to visual mode.
5. Delete the newline.
6. Switch to HTML mode and make sure the nodes were properly merged.
